cbqueryx/queryrespreader.go

```diff
@@ -122,27 +125,28 @@ func (r *queryRespReader) parseError(errJson *queryErrorJson) *ServerError {
 	if errCodeGroup == 5 {
 		err = ErrInternalServerError
 	}
-	if errCodeGroup == 12 || errCodeGroup == 14 && errCode != 12004 && errCode != 12016 {
+	if errCodeGroup == 12 || errCodeGroup == 14 {
```

Comment:
Are these custom couchbase server error codes?

Reply:
They're codes specific to the query service, and are included as a part of the "errors" field in the response payload.